src/components/Tag/Tag.tsx

```diff
@@ -1,72 +1,115 @@
 import clsx from 'clsx';
 import React from 'react';
-import type { ReactNode } from 'react';
-import type { IconName } from '../Icon';
-import Icon from '../Icon';
+import { assertEdsUsage } from '../../util/logging';
+import type { Status } from '../../util/variant-types';
+
+import Icon, { type IconName } from '../Icon';
 import Text from '../Text';
 
 import styles from './Tag.module.css';
 
+// TODO(next-major): remove these local variants and usages in code and stories
 export const VARIANTS = [
   'neutral',
   'error',
   'success',
   'warning',
   'brand',
 ] as const;
-
 export type Variant = (typeof VARIANTS)[number];
 
 type Props = {
-  /**
-   * The color variant of the tag. It will update the content colors, background color, and border (when `hasOutline` is set to `true`).
-   *
-   * **Default is `"neutral"`**.
-   */
-  variant?: Variant;
```

Comment:
Do we allow for breaking API contract with major version bump?

Reply:
Yep, we do. Here, I moved `variant` to another part of `Props` for consistency (and to alphabetize better). those TODO(next-major) comments are for when we do the next major, likely in the new year. 

This one is backwards-compatible with some warnings (those `assertEDSUsage` log messages)

Reply:
OK got it, thanks